File: backend-.NET/Vehiclecfg_group9/Vehiclecfg_group9/Controllers/SegmentController.cs
using Microsoft.AspNetCore.Mvc;
using Vehiclecfg_group9.Models;
using Vehiclecfg_group9.Services;

namespace Vehiclecfg_group9.Controllers
{
    [ApiController]
    [Route("api")]
    public class SegmentController : ControllerBase
    {
        private readonly SegmentService _service;

        public SegmentController(SegmentService service)
        {
            _service = service;
        }

        // GET api/getSegment
        [HttpGet("getSegment")]
        public async Task<IActionResult> GetAll()
        {
            var segments = await _service.GetAllAsync();
            return Ok(segments);
        }

        // POST api/saveSegment
        [HttpPost("saveSegment")]
        public async Task<IActionResult> SaveSegment([FromBody] SegmentMaster segment)
        {
            var savedSegment = await _service.AddSegmentAsync(segment);
            return Created("", savedSegment);
        }

        // PUT api/update/1
        [HttpPut("update/{id}")]
        public async Task<IActionResult> UpdateSegment(int id, [FromBody] SegmentMaster segment)
        {
            var updatedSegment = await _service.UpdateSegmentAsync(id, segment);
            return Ok(updatedSegment);
        }

        // DELETE api/delete/1
        [HttpDelete("delete/{id}")]
        public async Task<IActionResult> DeleteSegment(int id)
        {
            await _service.DeleteSegmentAsync(id);
            return NoContent();
        }
    }
}

File: CDAC_PROJECT/JavaBackend/src/main/java/com/vehiclecfg/controller/SegmentController.java
package com.vehiclecfg.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import com.vehiclecfg.entities.Segment;
import com.vehiclecfg.services.SegmentService;

@RestController
@RequestMapping("/api")
@CrossOrigin(origins = {
        "http://localhost:5173",
        "http://localhost:3000"
})
public class SegmentController {

    @Autowired
    private SegmentService service;

    // ==================== GET ALL ====================

    @GetMapping("/getSegment")
    public ResponseEntity<List<Segment>> getAll() {

        return ResponseEntity.ok(
                service.GetAll()
        );

    }

    // ==================== CREATE ====================

    @PostMapping("/saveSegment")
    public ResponseEntity<Segment> saveSegment(
            @RequestBody Segment segment) {

        Segment savedSegment =
                service.AddSegment(segment);

        return ResponseEntity.status(HttpStatus.CREATED)
                .body(savedSegment);

    }

    // ==================== UPDATE ====================

    @PutMapping("/update/{id}")
    public ResponseEntity<Segment> updateSegment(
            @PathVariable int id,
            @RequestBody Segment segment) {

        Segment updatedSegment =
                service.UpdateSegment(id, segment);

        return ResponseEntity.ok(updatedSegment);

    }

    // ==================== DELETE ====================

    @DeleteMapping("/delete/{id}")
    public ResponseEntity<Void> deleteSegment(
            @PathVariable int id) {

        service.DeleteSegment(id);

        return ResponseEntity.noContent().build();

    }

}